mypy/build.py

```diff
@@ -998,6 +999,117 @@ def generate_deps_for_cache(manager: BuildManager,
     return rdeps
 
 
+def save_baseline(manager: BuildManager, sources: List[BuildSource]) -> None:
+    if not manager.options.baseline_file or (
+        (not manager.options.write_baseline and not manager.options.auto_baseline)
+        or (not manager.options.write_baseline and manager.options.auto_baseline and (
+            not manager.errors.num_messages()
+            or manager.errors.baseline_sources != [source.path for source in sources]
+        ))
+    ):
+        # Indicate that writing was canceled
+        manager.options.write_baseline = False
+        return
+    new_baseline = manager.errors.prepare_baseline_errors()
+    file = Path(manager.options.baseline_file)
+    if not new_baseline:
+        if file.exists():
+            file.unlink()
+            print("No errors, baseline file removed")
+        else:
+            print("No errors, no baseline to write")
+        # Indicate that writing was canceled
+        manager.options.write_baseline = False
+        return
+    if new_baseline == manager.errors.original_baseline:
+        return
+    if not file.parent.exists():
+        file.parent.mkdir(parents=True)
+    data: Mapping[str, object]
+    if (
+        manager.options.baseline_format == "1.3"
```

Comment:
should this be a tuple like `version_info` so you can do checks like `if baseline_format >= (1, 3)`

Reply:
it could be, but this pr is getting feature creep already, I'll add a TODO comment